modules/rules_foreign_cc/0.7.1/patches/label.patch

```diff
@@ -0,0 +1,159 @@
+index 1d0ad29..cc15a55 100644
+--- a/foreign_cc/built_tools/make_build.bzl
++++ b/foreign_cc/built_tools/make_build.bzl
+@@ -103,7 +103,7 @@ make_tool = rule(
+     output_to_genfiles = True,
+     implementation = _make_tool_impl,
+     toolchains = [
+-        str(Label("//foreign_cc/private/framework:shell_toolchain")),
++        "@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain",
+         "@bazel_tools//tools/cpp:toolchain_type",
+     ],
+ )
+diff --git a/foreign_cc/built_tools/ninja_build.bzl b/foreign_cc/built_tools/ninja_build.bzl
+index 7127fc1..f21aea9 100644
+--- a/foreign_cc/built_tools/ninja_build.bzl
++++ b/foreign_cc/built_tools/ninja_build.bzl
+@@ -33,7 +33,7 @@ ninja_tool = rule(
+     output_to_genfiles = True,
+     implementation = _ninja_tool_impl,
+     toolchains = [
+-        str(Label("//foreign_cc/private/framework:shell_toolchain")),
++        "@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain",
+         "@bazel_tools//tools/cpp:toolchain_type",
+     ],
+ )
+diff --git a/foreign_cc/private/framework/toolchain.bzl b/foreign_cc/private/framework/toolchain.bzl
+index 01778d3..feb6b74 100644
+--- a/foreign_cc/private/framework/toolchain.bzl
++++ b/foreign_cc/private/framework/toolchain.bzl
+@@ -17,7 +17,7 @@ foreign_cc_framework_toolchain(
+ toolchain(
+     name = "toolchain",
+     toolchain_type = "@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain",
+-    toolchain = "//:commands",
++    toolchain = ":commands",
+     exec_compatible_with = {exec_compat},
+     target_compatible_with = {target_compat},
+ )
+@@ -51,14 +51,8 @@ def _framework_toolchain_repository_impl(repository_ctx):
+         repository_ctx (repository_ctx): The rule's context object
+     """
+ 
+-    # Ensure we always have an absolute label. This may not be the case
+-    # when building within the `@rules_foreign_cc` workspace.
+-    absolute_label = str(repository_ctx.attr.commands_src)
+-    if not absolute_label.startswith("@"):
+-        absolute_label = "@rules_foreign_cc" + absolute_label
+-
+     repository_ctx.file("defs.bzl", _DEFS_BZL_CONTENT.format(
+-        commands_src = absolute_label,
++        commands_src = repository_ctx.attr.commands_src,
+         symbols = "\n    ".join(["\"{}\",".format(symbol) for symbol in PLATFORM_COMMANDS.keys()]),
+         commands = "\n    ".join(["{cmd} = {cmd},".format(cmd = symbol) for symbol in PLATFORM_COMMANDS.keys()]),
+     ))
+@@ -72,9 +66,8 @@ framework_toolchain_repository = repository_rule(
+     doc = "A repository rule which defines a `@rules_foreign_cc//foreign_cc/private/framework:shell_toolchain` toolchain.",
+     implementation = _framework_toolchain_repository_impl,
+     attrs = {
+-        "commands_src": attr.label(
+-            doc = "The label of a `.bzl` source which defines toolchain commands",
+-            allow_files = [".bzl"],
++        "commands_src": attr.string(
```

Comment:
this is very brittle and won't work in most cases due to repo mapping (but there's probably no good workaround right now; see bazelbuild/bazel#14580).

EDIT: On second reading, it seems that this is a private repo rule, only ever called from rules_foreign_cc's own module extension. In that case, you know that the generated repo will have visibility into the rules_foreign_cc repo under the name `rules_foreign_cc`, so everything will probably actually (happen to) work fine. Nevertheless, it still seems brittle; the issue linked above should be the ultimate solution.